### ISSUE TITLE
AbstractMethodError when testing a class with a (static) com.google.common.reflect.TypeToken field

### ISSUE THREAD

**Issue by ylexus** [jqno/equalsverifier#238]
# What steps will reproduce the problem?
Run EqualsVerifier on the following class:
```
import com.google.common.reflect.TypeToken;

/**
 * Command for handling events
 */
final class MyClass {
    private static final TypeToken<?> TYPE_TOKEN = new TypeToken<Object>() {};

    private final Object argument;

    private MyClass(Object argument) {
        this.argument = argument;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }

        MyClass that = (MyClass) obj;

        return argument.equals(that.argument);
    }

    @Override
    public int hashCode() {
        return argument.hashCode();
    }
}
```

# What is the code that triggers this problem?
```
        EqualsVerifier.forClass(MyClass.class)
                .suppress(Warning.NULL_FIELDS)
                .verify();
```

Provide an example of a complete class (equals method, hashCode method, relevant fields) and a call to EqualsVerifier.
see above

# What error message or stack trace does EqualsVerifier give?
```
java.lang.AssertionError: EqualsVerifier found a problem in class MyClass.
-> Method nl/jqno/equalsverifier/internal/reflection/TypeVariable$$DynamicSubclass$1790585734.getGenericDeclaration()Ljava/lang/reflect/GenericDeclaration; is abstract

For more information, go to: http://www.jqno.nl/equalsverifier/errormessages

	at nl.jqno.equalsverifier.EqualsVerifierApi.verify(EqualsVerifierApi.java:308)
	at com.baml.gmt.xasf.container.dispatch.impl.MyClassTest.testEquals(MyClassTest.java:12)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.AbstractMethodError: Method nl/jqno/equalsverifier/internal/reflection/TypeVariable$$DynamicSubclass$1790585734.getGenericDeclaration()Ljava/lang/reflect/GenericDeclaration; is abstract
	at nl.jqno.equalsverifier.internal.reflection.TypeVariable$$DynamicSubclass$1790585734.getGenericDeclaration(Unknown Source)
	at com.google.common.reflect.TypeResolver$TypeVariableKey.hashCode(TypeResolver.java:561)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.put(HashMap.java:612)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.MapFactory.createValues(MapFactory.java:37)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.createTuple(PrefabValues.java:160)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.realizeCacheFor(PrefabValues.java:147)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.CopyFactory.createValues(CopyFactory.java:23)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.ExternalFactory.createValues(ExternalFactory.java:34)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.createTuple(PrefabValues.java:160)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.realizeCacheFor(PrefabValues.java:147)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.traverseFields(FallbackFactory.java:77)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.createValues(FallbackFactory.java:36)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.createTuple(PrefabValues.java:164)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.realizeCacheFor(PrefabValues.java:147)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.traverseFields(FallbackFactory.java:77)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.createValues(FallbackFactory.java:36)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.createTuple(PrefabValues.java:164)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.realizeCacheFor(PrefabValues.java:147)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.traverseFields(FallbackFactory.java:77)
	at nl.jqno.equalsverifier.internal.prefabvalues.factories.FallbackFactory.createValues(FallbackFactory.java:36)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.createTuple(PrefabValues.java:164)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.realizeCacheFor(PrefabValues.java:147)
	at nl.jqno.equalsverifier.internal.prefabvalues.PrefabValues.giveTuple(PrefabValues.java:89)
	at nl.jqno.equalsverifier.internal.checkers.AbstractDelegationChecker.safelyGetTuple(AbstractDelegationChecker.java:73)
	at nl.jqno.equalsverifier.internal.checkers.AbstractDelegationChecker.checkAbstractDelegationInFields(AbstractDelegationChecker.java:62)
	at nl.jqno.equalsverifier.internal.checkers.AbstractDelegationChecker.check(AbstractDelegationChecker.java:36)
	at nl.jqno.equalsverifier.EqualsVerifierApi.verifyWithoutExamples(EqualsVerifierApi.java:368)
	at nl.jqno.equalsverifier.EqualsVerifierApi.performVerification(EqualsVerifierApi.java:349)
	at nl.jqno.equalsverifier.EqualsVerifierApi.verify(EqualsVerifierApi.java:302)
	... 23 more
```

# What did you expect?
A successful execution.

# Which version of EqualsVerifier are you using?
3.1.7
(worked in 2.0.1)

**Comment by jqno** [comment]
Reproduced. I will fix this for the next version, I'll let you know when I release it.
In the mean time, you can use `withPrefabValues` as a workaround.

**Comment by jqno** [comment]
I've just released version 3.1.8 which fixes this.
